cubes/new_cube.py

```diff
@@ -0,0 +1,254 @@
+""" JXP VERSION """
+
+'''
+Questions for Clancy:
+
+1. lC vs. C
+2. Order of update_grid()
```

Comment:
This will require a detailed review. Provided changing H0 is kept as an outermost loop, naively changing everything for H0 (i.e. effectively creating a new grid, albeit in such a way that python does not leak memory - oh, for c++! - won't lose much time). In the future we should aim to improve speed here. Also for evaluating the energy function - there are opportunities to use more memory and save time in the long run.